apis/placement/v1beta1/clusterresourceplacement_types.go

```diff
@@ -657,53 +657,78 @@ const (
 type ResourcePlacementStatus struct {
 	// ClusterName is the name of the cluster this resource is assigned to.
 	// If it is not empty, its value should be unique cross all placement decisions for the Placement.
-	// +optional
+	// +kubebuilder:validation:Optional
 	ClusterName string `json:"clusterName,omitempty"`
 
 	// ApplicableResourceOverrides contains a list of applicable ResourceOverride snapshots associated with the selected
 	// resources.
 	//
 	// This field is alpha-level and is for the override policy feature.
-	// +optional
+	// +kubebuilder:validation:Optional
 	ApplicableResourceOverrides []NamespacedName `json:"applicableResourceOverrides,omitempty"`
 
 	// ApplicableClusterResourceOverrides contains a list of applicable ClusterResourceOverride snapshots associated with
 	// the selected resources.
 	//
 	// This field is alpha-level and is for the override policy feature.
-	// +optional
+	// +kubebuilder:validation:Optional
 	ApplicableClusterResourceOverrides []string `json:"applicableClusterResourceOverrides,omitempty"`
 
 	// +kubebuilder:validation:MaxItems=100
 
 	// FailedPlacements is a list of all the resources failed to be placed to the given cluster or the resource is unavailable.
 	// Note that we only include 100 failed resource placements even if there are more than 100.
 	// This field is only meaningful if the `ClusterName` is not empty.
-	// +optional
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:MaxItems=100
 	FailedPlacements []FailedResourcePlacement `json:"failedPlacements,omitempty"`
 
+	// DriftedPlacements is a list of all the resources whose label/annotation and spec fields have changed on the
+	// target cluster after it is successfully applied.
+	// This field is only meaningful if the `ClusterName` is not empty.
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:MaxItems=100
+	DriftedPlacements []DriftedResourcePlacement `json:"driftedPlacements,omitempty"`
+
 	// Conditions is an array of current observed conditions for ResourcePlacementStatus.
-	// +optional
+	// +kubebuilder:validation:Optional
 	Conditions []metav1.Condition `json:"conditions,omitempty"`
 }
 
 // FailedResourcePlacement contains the failure details of a failed resource placement.
 type FailedResourcePlacement struct {
 	// The resource failed to be placed.
-	// +required
+	// +kubebuilder:validation:Required
 	ResourceIdentifier `json:",inline"`
 
 	// The failed condition status.
-	// +required
+	// +kubebuilder:validation:Required
 	Condition metav1.Condition `json:"condition"`
 }
 
+// DriftedResourcePlacement contains the drifted resource details.
+type DriftedResourcePlacement struct {
+	// The resource that has drifted.
+	// +kubebuilder:validation:Required
+	ResourceIdentifier `json:",inline"`
+
+	// The ObservedGeneration of the resource on the target cluster that caused this drift.
+	// +kubebuilder:validation:Required
+	ObservedGeneration int64 `json:"observedGeneration"`
```

Comment:
Hi Ryan! Just a quick question: this assumes that we will be switching to our own condition implementations to relay this piece of information back, right?

Reply:
Also, considering that `observedGeneration` is kind of a well-known name in the K8s community, do we want to be more specific that it's about the generation on the target cluster? This is just a nit though, of course.

Reply:
Just a random thought, but a `driftedSince` field could also be useful, just to pinpoint the cause

Reply:
Add observedAt to denote when the observation is made?